realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -0,0 +1,153 @@
+#include <object-store/src/object_store.hpp>
+#include "io_realm_internal_SharedRealm.h"
+
+#include "shared_realm.hpp"
+#include "util.hpp"
+
+using namespace realm;
+
+static std::vector<char> jbytearray_to_vector(JNIEnv *env, jbyteArray bytes) {
+    std::vector<char> v(env->GetArrayLength(bytes));
+    jbyte *ptr = env->GetByteArrayElements(bytes, NULL);
+    std::copy_n(ptr, v.size(), v.begin());
+    env->ReleaseByteArrayElements(bytes, ptr, JNI_ABORT);
+    return v;
+}
+
+JNIEXPORT jlong JNICALL
+Java_io_realm_internal_SharedRealm_nativeCreateConfig
+(JNIEnv *env, jclass, jstring realm_path, jbyteArray key, jboolean read_only, jboolean in_memory,
+ jboolean cache, jboolean disable_format_upgrade, jboolean auto_change_notification) {
+    TR_ENTER()
+    JStringAccessor path(env, realm_path);
+    Realm::Config *config = new Realm::Config();
+    config->path = path;
+    if (key != nullptr) {
+        config->encryption_key = jbytearray_to_vector(env, key);
+    }
+    config->read_only = read_only;
+    config->in_memory = in_memory;
+    config->cache = cache;
+    config->disable_format_upgrade = disable_format_upgrade;
+    config->automatic_change_notifications = auto_change_notification;
+    return reinterpret_cast<jlong>(config);
+}
+
+JNIEXPORT void JNICALL
+Java_io_realm_internal_SharedRealm_nativeCloseConfig
+(JNIEnv *, jclass, jlong config_ptr) {
+    TR_ENTER_PTR(config_ptr)
+    delete RC(config_ptr);
+}
+
+JNIEXPORT jlong JNICALL
+Java_io_realm_internal_SharedRealm_nativeGetSharedRealm
+(JNIEnv *, jclass, jlong config_ptr) {
+    TR_ENTER_PTR(config_ptr)
+    SharedRealm shared_realm = Realm::get_shared_realm(*RC(config_ptr));
+    return reinterpret_cast<jlong>(new SharedRealm(shared_realm));
+}
+
+JNIEXPORT void JNICALL
+Java_io_realm_internal_SharedRealm_nativeCloseSharedRealm
+(JNIEnv *, jclass, jlong shared_realm_ptr) {
+    TR_ENTER_PTR(shared_realm_ptr)
+    delete SR(shared_realm_ptr);
```

Comment:
`SharedRealm` is actually `std::shared_ptr<Realm`.
And we are not holding a pointer to `Realm`, instead, we are holding a pointer to `shared_ptr`.

So this will delete the `share_ptr` created in line 48. The one created/got in line 47 may/may not be deleted, since it depends one how many `shared_ptr` still point to the `Realm`.